app/models/evaluation.rb

```diff
@@ -75,6 +76,6 @@ def calculate_total_score
   end
 
   def update_submission_evaluation_status
-    submission.update_submission_evaluation_status
+    submission.save
```

Comment:
call the Service directly instead of relying on the before_save callback:
```suggestion
    EvaluationStatusService.update_evaluation_status(submission)
```